qutebrowser/config/configtypes.py

```diff
@@ -1029,17 +1029,27 @@ def to_py(self, value: _StrUnset) -> typing.Union[configutils.Unset,
             openparen = value.index('(')
             kind = value[:openparen]
             vals = value[openparen+1:-1].split(',')
-            int_vals = [self._parse_value(v) for v in vals]
-            if kind == 'rgba' and len(int_vals) == 4:
-                return QColor.fromRgb(*int_vals)
-            elif kind == 'rgb' and len(int_vals) == 3:
-                return QColor.fromRgb(*int_vals)
-            elif kind == 'hsva' and len(int_vals) == 4:
-                return QColor.fromHsv(*int_vals)
-            elif kind == 'hsv' and len(int_vals) == 3:
-                return QColor.fromHsv(*int_vals)
-            else:
-                raise configexc.ValidationError(value, "must be a valid color")
+
+            converters = {
+                'rgba': QColor.fromRgb,
+                'rgb': QColor.fromRgb,
+                'hsva': QColor.fromHsv,
+                'hsv': QColor.fromHsv,
+            }
```

Comment:
```suggestion
            }  # type: typing.Dict[str, typing.Callable[..., QColor]]
```

@jgkamat instead proposed:

```suggestion
            }  # type: typing.Dict[str, typing.Callable[[int, int, int], QColor]]
```

which is more strict, but not really correct, as the callable also can take four arguments. Since we probably can't really  convince mypy the rest of the code is correct (because we check the number of arguments dynamically) let's just be a bit more permissive here.